driver/include/i2c.hpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file include/i2c.hpp
+ * @brief Drivers for I2C using the kernel smbus driver.
+ */
+#pragma once
+
+class i2c {
+
+public:
+
+    /**
+     * @brief Constructor. Initializes I2C device.
+     *
+     * @param[in] dev_addr - address of the I2C slave device.
+     */
+    i2c(const uint8_t dev_addr);
+
+    /** 
+     * @brief I2C read.
+     *
+     * @param[in] data - Data buffer to write to.
+     * @param[in] size - Size of data buffer.
+     * @return error code
+     */
+    int read(char* data, uint8_t size);
```

Comment:
Return the size of the read buffer

read/write functions should be marked const